CONTRIBUTING_GO.md

```diff
@@ -161,6 +164,45 @@ reasons why this is _the right thing to do_:
 ```
+
+## Contributor Principles
+
+These principles are a living document and should be updated as-needed to better serve the go-ipfs community.
+
+In order of precedence:
+
+1. **Build the distributed peer-to-peer web.** This is our objective and aim - everything we do is in service of this goal and we only compromise our other principles if needed to achieve this.
+2. **Don't break the IPFS ecosystem.** This isn't "don't break protocols"; we do occasionally break protocols and APIs (with care) when we know that users won't notice it. However, users trust us to not break their workflows/projects. If we do, we'll lose that trust and those users.
+3. **Take ownership.** If you're working on something, own it and see it through. Given that (a) we don't have managers and (b) we don't work synchronously, coordination can be difficult. To make up for this, if you want something to happen, you have to make it happen (that may mean calling a meeting, bugging people, fixing unrelated issues, designing alternative solutions, etc.).
+4. **Ask for forgiveness, not permission.** To enable P.3, if you're having trouble getting people to move on something, err on the side of getting things done (bounded by P.1 and P.2). Note: This doesn't apply when you *know* you'll need to ask for forgiveness... use your best judgment.
+
+## Code Change Policy
+
+The go code change policy must balance "not breaking things" with "getting things done" and should be read in that spirit. Use your best judgment when making changes and PRs; these rules are not set in stone.
+
+**How can the codebase be changed?**
+
+Except in special cases (and mass gx-workspace updates), all changes go through PRs.
+
+* Larger changes should be discussed in an issue *before* work is started to avoid wasted effort, tears, and dead fairies.
+* WIP (work in progress), RFC (request for comments) changes should be marked as such. Not doing so can rouse the review gremlins.
+
+**Who can merge a PR?**
+
+Changes should be merged by, in-order of preference:
+
+* The repo's maintainer (defacto or otherwise).
+* The PR author (if they have write access).
+* A PR reviewer (someone who has reviewed the *entire* PR).
+
+In general, a PR should be merged by someone that knows it's ready and *should not* be merged by someone just because they see that it has been approved. However, use your discretion: reviewers (with write access) should feel free to merge small bug fixes, etc. immediately.
+
+**When can a PR be merged?**
+
+* Small changes should get one review from someone (preferably on the go team if it's a code change) who can sanity check the change.
+* Large changes, especially interface changes, should get signoffs from anyone with a "stake". This is *usually* just the reviewers that GitHub recommends. However, if you're not sure, say so and let someone else merge perform the actual merge. Never be afraid to ask for guidance.
+* All merges to go-ipfs are gated by the maintainer in service of P.2.
```

Comment:
I'm confused - you give two other ways for merges to be performed, and then you say they're all gated by the maintainer. When/how does that happen if other reviewers are encouraged to merge if they have write access?

Reply:
To go-ipfs and only go-ipfs. I should probably say "to go-ipfs *itself*".

Reply:
Still doesn't clarify to me whether as PR author I can actually merge small changes, or as a reviewer I can merge large changes - if the lead maintainer is the gatekeeper. Do you just mean that the maintainer can revert any merged changes, or that the go-ipfs maintainer is the final approver on any/all changes?

Reply:
Yeah, that's still really confusing. Does my latest change clarify this?

Reply:
MUCH CLEARER! Thank you.